Source/SwiftLintFramework/Protocols/Rule.swift

```diff
@@ -32,11 +37,32 @@ public protocol ConfigurationProviderRule: Rule {
 }
 
 public protocol CorrectableRule: Rule {
+    func correct(file: File, compilerArguments: [String]) -> [Correction]
     func correct(file: File) -> [Correction]
 }
 
+public extension CorrectableRule {
+    func correct(file: File, compilerArguments: [String]) -> [Correction] {
+        return correct(file: file)
+    }
+}
+
 public protocol SourceKitFreeRule: Rule {}
 
+public protocol AnalyzerRule: Rule {}
```

Comment:
should it inherit from `OptInRule` for now?

Reply:
Good call, done.